src/python/pants/jvm/target_types.py

```diff
@@ -517,7 +517,7 @@ async def generate_from_pom_xml(
     )
     files = await Get(DigestContents, Digest, pom_xml.snapshot.digest)
     mapping = request.generator[JvmArtifactsPackageMappingField].value
-    coordinates = parse_pom_xml(files[0].content)
+    coordinates = parse_pom_xml(files[0].content, pom_xml_path=pom_xml.snapshot.files[0])
```

Comment:
It just occurred to me that if `files` is empty (which it shouldn't be due to other checks, but I hate relying on non-local checks) this will crash with a non-useful error message. So I'd recommend checking `len(files)` and erroring with "Found no file at <path>"

Reply:
Sure https://github.com/pantsbuild/pants/pull/20336/commits/b68ec95ee944d7366621022d49fdb24a2e1f4af8